### PR TITLE
rewrite eventbus subscribe: add message handler

### PR DESCRIPTION
**Pull request by Shelley-BaoYue**
Signed-off-by: Shelley-BaoYue <baoyue2@huawei.com>

**What type of PR is this?**
/kind design

**What this PR does / why we need it**:
for eventbus scalability，rewirte eventbus subscribe part, add msg handler for each topic

### CODE REVIEW

File: edge/pkg/eventbus/mqtt/filter.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+	pattern     *MessagePattern
+	handlerFunc HandlerFunc
+}
+
+// MessagePattern message pattern
+type MessagePattern struct {
+	resource  string
+	operation string
+
+	resExpr *MessageExpression
```
Comment (vincentgoat) [inline edge/pkg/eventbus/mqtt/filter.go:25]: hi @Shelley-BaoYue, move the value of the pointer type to the first line of the struct?

File: edge/pkg/eventbus/mqtt/filter.go [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+}
+
+// MessageExpression message expression
+type MessageExpression struct {
+	VarNames []string
+	VarCount int
+	Matcher  *regexp.Regexp
```
Comment (vincentgoat) [inline edge/pkg/eventbus/mqtt/filter.go:32]: the same as above

File: edge/pkg/eventbus/mqtt/filter.go [hunk 3]
```diff
@@ -0,0 +1,158 @@
+package mqtt
```
Comment (wackxu) [inline edge/pkg/eventbus/mqtt/filter.go:1]: we should add some unit tests for the filter.go

File: edge/pkg/eventbus/mqtt/filter.go (hunk 1 above)
Comment (Shelley-BaoYue) [reply edge/pkg/eventbus/mqtt/filter.go]: > hi @Shelley-BaoYue, move the value of the pointer type to the first line of the struct?

done

File: edge/pkg/eventbus/mqtt/filter.go (hunk 3 above)
Comment (Shelley-BaoYue) [reply edge/pkg/eventbus/mqtt/filter.go:1]: > we should add some unit tests for the filter.go

done with unit test

File: edge/pkg/eventbus/mqtt/filter.go [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+		} else {
+			buffer.WriteString(regexp.QuoteMeta(field))
+		}
+		buffer.WriteString("/")
+	}
+
+	expression := strings.TrimRight(buffer.String(), "/") + "(/.*)?$"
+	compiled, err := regexp.Compile(expression)
+	if err != nil {
+		klog.Errorf("failed to compile resource with error %+v", err)
```
Comment (fisherxu) [inline edge/pkg/eventbus/mqtt/filter.go:84]:
```suggestion
		klog.Errorf("failed to compile resource, error: %+v", err)
```

File: edge/pkg/eventbus/mqtt/filter.go [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+func NewMessageMux() *MessageMux {
+	return &mux
+}
+
+// NewPattern new pattern
+func NewPattern(resource string) *MessagePattern {
+	expression := NewExpression()
+	resExpr := expression.GetExpression(resource)
+	if resExpr == nil {
+		klog.Errorf("bad resource for expression:%s", resource)
```
Comment (fisherxu) [inline edge/pkg/eventbus/mqtt/filter.go:107]:
```suggestion
		klog.Errorf("bad resource for expression: %s", resource)
```

File: edge/pkg/eventbus/mqtt/filter_test.go [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+		QOS:     0,
+		Retain:  false,
+	}
+	NewMessageMux().Dispatch(msg.Topic, msg.Payload)
+	message, _ := beehiveContext.Receive(modules.DeviceTwinModuleName)
+
+	t.Run("SuccessDispatchDeviceTwinMsg", func(t *testing.T) {
+		want := messagepkg.OperationResponse
+		if message.GetOperation() != want {
+			t.Errorf("Wrong message received : Wanted operation %v and Got operation %v", want, message.GetOperation())
```
Comment (fisherxu) [inline edge/pkg/eventbus/mqtt/filter_test.go:39]:
```suggestion
			t.Errorf("Wrong message received : Wanted operation: %v and Got operation: %v", want, message.GetOperation())
```

File: edge/pkg/eventbus/mqtt/filter.go [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+	}
+	handleUploadTopic(topic, payload)
+}
+
+// RegisterMsgHandler register handler for message if topic is matched in pattern
+// for "$hw/events/device/+/twin/+", "$hw/events/node/+/membership/get", send to twin
+// for other, send to hub
+// for "SYS/dis/upload_records", no need to base64 topic
+func RegisterMsgHandler() {
+	mux.Entry(NewPattern("$hw/events/device/"), handleDeviceTwin)
```
Comment (zc2638) [inline edge/pkg/eventbus/mqtt/filter.go:155]: Is it appropriate to change `$hw` to `$ke`? There can be a version-compatible process, and `$ke` should prevail in the future

Comment (fisherxu) [reply edge/pkg/eventbus/mqtt/filter.go:155]: +1, and let's do this in another PR :) This pr is focus on the refactor

**Review by fisherxu — COMMENTED** [review]
/lgtm
/hold cancel

**Pull request merged**